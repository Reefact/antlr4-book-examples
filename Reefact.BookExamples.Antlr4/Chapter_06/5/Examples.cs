#region Usings declarations

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_06._5 {

    [UseReporter(typeof(BeyondCompareReporter))]
    public class Examples {

        [Fact]
        [GraphicalTree("ParseTree1.svg")]
        public void mermaid_style_tree() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("t.R", 6, 5);
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        [GraphicalTree("ParseTree2.svg")]
        public void ajay_shah_b1_example() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("b1.R", 6, 5);
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        public void ajay_shah_b5_example() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("b5.R", 6, 5);
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string tokensString = grun.ToTokensString();
            // Verify
            Approvals.Verify(tokensString);
        }

    }

}