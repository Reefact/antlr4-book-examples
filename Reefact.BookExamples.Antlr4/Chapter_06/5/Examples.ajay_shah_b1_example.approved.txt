graph TD
	1["prog"] --> 2["# Ajay Shah b1 example (https://www.mayin.org/ajayshah/KB/R/html/b1.html)#0092;r#0092;n"]
	1 --> 3["# Goals: A first look at R objects - vectors, lists, matrices, data frames.#0092;r#0092;n"]
	1 --> 4["#0092;r#0092;n"]
	1 --> 5["# To make vectors #0034;x#0034; #0034;y#0034; #0034;year#0034; and #0034;names#0034;#0092;r#0092;n"]
	1 --> 6["expr_or_assign"]
	6 --> 7["expr"]
	7 --> 8["x"]
	6 --> 9["#0060;-"]
	6 --> 10["expr_or_assign"]
	10 --> 11["expr"]
	11 --> 12["expr"]
	12 --> 13["c"]
	11 --> 14["("]
	11 --> 15["sublist"]
	15 --> 16["sub"]
	16 --> 17["expr"]
	17 --> 18["2"]
	15 --> 19[","]
	15 --> 20["sub"]
	20 --> 21["expr"]
	21 --> 22["3"]
	15 --> 23[","]
	15 --> 24["sub"]
	24 --> 25["expr"]
	25 --> 26["7"]
	15 --> 27[","]
	15 --> 28["sub"]
	28 --> 29["expr"]
	29 --> 30["9"]
	11 --> 31[")"]
	1 --> 32["#0092;r#0092;n"]
	1 --> 33["expr_or_assign"]
	33 --> 34["expr"]
	34 --> 35["y"]
	33 --> 36["#0060;-"]
	33 --> 37["expr_or_assign"]
	37 --> 38["expr"]
	38 --> 39["expr"]
	39 --> 40["c"]
	38 --> 41["("]
	38 --> 42["sublist"]
	42 --> 43["sub"]
	43 --> 44["expr"]
	44 --> 45["9"]
	42 --> 46[","]
	42 --> 47["sub"]
	47 --> 48["expr"]
	48 --> 49["7"]
	42 --> 50[","]
	42 --> 51["sub"]
	51 --> 52["expr"]
	52 --> 53["3"]
	42 --> 54[","]
	42 --> 55["sub"]
	55 --> 56["expr"]
	56 --> 57["2"]
	38 --> 58[")"]
	1 --> 59["#0092;r#0092;n"]
	1 --> 60["expr_or_assign"]
	60 --> 61["expr"]
	61 --> 62["year"]
	60 --> 63["#0060;-"]
	60 --> 64["expr_or_assign"]
	64 --> 65["expr"]
	65 --> 66["expr"]
	66 --> 67["1990"]
	65 --> 68[":"]
	65 --> 69["expr"]
	69 --> 70["1993"]
	1 --> 71["#0092;r#0092;n"]
	1 --> 72["expr_or_assign"]
	72 --> 73["expr"]
	73 --> 74["names"]
	72 --> 75["#0060;-"]
	72 --> 76["expr_or_assign"]
	76 --> 77["expr"]
	77 --> 78["expr"]
	78 --> 79["c"]
	77 --> 80["("]
	77 --> 81["sublist"]
	81 --> 82["sub"]
	82 --> 83["expr"]
	83 --> 84["#0034;payal#0034;"]
	81 --> 85[","]
	81 --> 86["sub"]
	86 --> 87["expr"]
	87 --> 88["#0034;shraddha#0034;"]
	81 --> 89[","]
	81 --> 90["sub"]
	90 --> 91["expr"]
	91 --> 92["#0034;kritika#0034;"]
	81 --> 93[","]
	81 --> 94["sub"]
	94 --> 95["expr"]
	95 --> 96["#0034;itida#0034;"]
	77 --> 97[")"]
	1 --> 98["#0092;r#0092;n"]
	1 --> 99["# Accessing the 1st and last elements of y --#0092;r#0092;n"]
	1 --> 100["expr_or_assign"]
	100 --> 101["expr"]
	101 --> 102["expr"]
	102 --> 103["y"]
	101 --> 104["["]
	101 --> 105["sublist"]
	105 --> 106["sub"]
	106 --> 107["expr"]
	107 --> 108["1"]
	101 --> 109["]"]
	1 --> 110["#0092;r#0092;n"]
	1 --> 111["expr_or_assign"]
	111 --> 112["expr"]
	112 --> 113["expr"]
	113 --> 114["y"]
	112 --> 115["["]
	112 --> 116["sublist"]
	116 --> 117["sub"]
	117 --> 118["expr"]
	118 --> 119["expr"]
	119 --> 120["length"]
	118 --> 121["("]
	118 --> 122["sublist"]
	122 --> 123["sub"]
	123 --> 124["expr"]
	124 --> 125["y"]
	118 --> 126[")"]
	112 --> 127["]"]
	1 --> 128["#0092;r#0092;n"]
	1 --> 129["#0092;r#0092;n"]
	1 --> 130["# To make a list #0034;person#0034; --#0092;r#0092;n"]
	1 --> 131["expr_or_assign"]
	131 --> 132["expr"]
	132 --> 133["person"]
	131 --> 134["#0060;-"]
	131 --> 135["expr_or_assign"]
	135 --> 136["expr"]
	136 --> 137["expr"]
	137 --> 138["list"]
	136 --> 139["("]
	136 --> 140["sublist"]
	140 --> 141["sub"]
	141 --> 142["name"]
	141 --> 143["="]
	141 --> 144["expr"]
	144 --> 145["#0034;payal#0034;"]
	140 --> 146[","]
	140 --> 147["sub"]
	147 --> 148["x"]
	147 --> 149["="]
	147 --> 150["expr"]
	150 --> 151["2"]
	140 --> 152[","]
	140 --> 153["sub"]
	153 --> 154["y"]
	153 --> 155["="]
	153 --> 156["expr"]
	156 --> 157["9"]
	140 --> 158[","]
	140 --> 159["sub"]
	159 --> 160["year"]
	159 --> 161["="]
	159 --> 162["expr"]
	162 --> 163["1990"]
	136 --> 164[")"]
	1 --> 165["#0092;r#0092;n"]
	1 --> 166["expr_or_assign"]
	166 --> 167["expr"]
	167 --> 168["person"]
	1 --> 169["#0092;r#0092;n"]
	1 --> 170["# Accessing things inside a list --#0092;r#0092;n"]
	1 --> 171["expr_or_assign"]
	171 --> 172["expr"]
	172 --> 173["expr"]
	173 --> 174["person"]
	172 --> 175["$"]
	172 --> 176["expr"]
	176 --> 177["name"]
	1 --> 178["#0092;r#0092;n"]
	1 --> 179["expr_or_assign"]
	179 --> 180["expr"]
	180 --> 181["expr"]
	181 --> 182["person"]
	180 --> 183["$"]
	180 --> 184["expr"]
	184 --> 185["x"]
	1 --> 186["#0092;r#0092;n"]
	1 --> 187["#0092;r#0092;n"]
	1 --> 188["# To make a matrix, pasting together the columns #0034;year#0034; #0034;x#0034; and #0034;y#0034;#0092;r#0092;n"]
	1 --> 189["# The verb cbind() stands for #0034;column bind#0034;#0092;r#0092;n"]
	1 --> 190["expr_or_assign"]
	190 --> 191["expr"]
	191 --> 192["expr"]
	192 --> 193["cbind"]
	191 --> 194["("]
	191 --> 195["sublist"]
	195 --> 196["sub"]
	196 --> 197["expr"]
	197 --> 198["year"]
	195 --> 199[","]
	195 --> 200["sub"]
	200 --> 201["expr"]
	201 --> 202["x"]
	195 --> 203[","]
	195 --> 204["sub"]
	204 --> 205["expr"]
	205 --> 206["y"]
	191 --> 207[")"]
	1 --> 208["#0092;r#0092;n"]
	1 --> 209["#0092;r#0092;n"]
	1 --> 210["# To make a #0034;data frame#0034;, which is a list of vectors of the same length --#0092;r#0092;n"]
	1 --> 211["expr_or_assign"]
	211 --> 212["expr"]
	212 --> 213["D"]
	211 --> 214["#0060;-"]
	211 --> 215["expr_or_assign"]
	215 --> 216["expr"]
	216 --> 217["expr"]
	217 --> 218["data.frame"]
	216 --> 219["("]
	216 --> 220["sublist"]
	220 --> 221["sub"]
	221 --> 222["expr"]
	222 --> 223["names"]
	220 --> 224[","]
	220 --> 225["sub"]
	225 --> 226["expr"]
	226 --> 227["year"]
	220 --> 228[","]
	220 --> 229["sub"]
	229 --> 230["expr"]
	230 --> 231["x"]
	220 --> 232[","]
	220 --> 233["sub"]
	233 --> 234["expr"]
	234 --> 235["y"]
	216 --> 236[")"]
	1 --> 237["#0092;r#0092;n"]
	1 --> 238["expr_or_assign"]
	238 --> 239["expr"]
	239 --> 240["expr"]
	240 --> 241["nrow"]
	239 --> 242["("]
	239 --> 243["sublist"]
	243 --> 244["sub"]
	244 --> 245["expr"]
	245 --> 246["D"]
	239 --> 247[")"]
	1 --> 248["#0092;r#0092;n"]
	1 --> 249["# Accessing one of these vectors#0092;r#0092;n"]
	1 --> 250["expr_or_assign"]
	250 --> 251["expr"]
	251 --> 252["expr"]
	252 --> 253["D"]
	251 --> 254["$"]
	251 --> 255["expr"]
	255 --> 256["names"]
	1 --> 257["#0092;r#0092;n"]
	1 --> 258["# Accessing the last element of this vector#0092;r#0092;n"]
	1 --> 259["expr_or_assign"]
	259 --> 260["expr"]
	260 --> 261["expr"]
	261 --> 262["D"]
	260 --> 263["$"]
	260 --> 264["expr"]
	264 --> 265["expr"]
	265 --> 266["names"]
	264 --> 267["["]
	264 --> 268["sublist"]
	268 --> 269["sub"]
	269 --> 270["expr"]
	270 --> 271["expr"]
	271 --> 272["nrow"]
	270 --> 273["("]
	270 --> 274["sublist"]
	274 --> 275["sub"]
	275 --> 276["expr"]
	276 --> 277["D"]
	270 --> 278[")"]
	264 --> 279["]"]
	1 --> 280["#0092;r#0092;n"]
	1 --> 281["# Or equally,#0092;r#0092;n"]
	1 --> 282["expr_or_assign"]
	282 --> 283["expr"]
	283 --> 284["expr"]
	284 --> 285["D"]
	283 --> 286["$"]
	283 --> 287["expr"]
	287 --> 288["expr"]
	288 --> 289["names"]
	287 --> 290["["]
	287 --> 291["sublist"]
	291 --> 292["sub"]
	292 --> 293["expr"]
	293 --> 294["expr"]
	294 --> 295["length"]
	293 --> 296["("]
	293 --> 297["sublist"]
	297 --> 298["sub"]
	298 --> 299["expr"]
	299 --> 300["expr"]
	300 --> 301["D"]
	299 --> 302["$"]
	299 --> 303["expr"]
	303 --> 304["names"]
	293 --> 305[")"]
	287 --> 306["]"]
	1 --> 307["#0060;EOF#0062;"]

classDef default fill:#fff,stroke:#000,stroke-width:0.25px;
classDef error color:#fff,fill:#FF0000,stroke:#000,stroke-width:0.25px;